src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -2017,6 +2017,72 @@ string[] linkingTargetFields
+        [DataRow("my-config.json", "Test", false, true, "my-config.Test.json", "my-config.json", DisplayName = "Config file in the current directory provided by user and environment variable is set and absolute path is provided")]
+        [DataRow("test-configs/my-config.json", "Test", true, true, "test-configs/my-config.Test.json", "test-configs/my-config.json", DisplayName = "Config file in different directory provided by user and environment variable is set and absolute path is provided")]
+        [DataRow("my-config.json", "Test", false, true, "dab-config.Test.json", "my-config.json", DisplayName = "Config file in the current directory provided by user and environment variable is set and environment file is present and absolute path is provided")]
+        [DataRow("test-configs/my-config.json", "Test", true, true, "test-configs/dab-config.Test.json", "test-configs/my-config.json", DisplayName = "Config file in the different directory provided by user and environment variable is set and environment file is present and absolute path is provided")]
+        [DataRow(null, "", false, false, null, "dab-config.json", DisplayName = "Config file not provided by user and environment variable is not set")]
+        [DataRow(null, "Test", false, false, "dab-config.Test.json", "dab-config.json", DisplayName = "Config file not provided by user and environment variable is set and environment file is present")]
+        [DataRow(null, "Test", false, false, null, "dab-config.json", DisplayName = "Config file not provided by user and environment variable is set and environment file is not present")]
+        public void TestCorrectConfigFileIsSelectedForRuntimeEngine(
+            string userProvidedConfigFilePath,
+            string environmentValue,
+            bool IsDifferentDirectory,
+            bool isAbsolutePath,
+            string environmentFile,
+            string finalConfigFilePath)
+        {
+            MockFileSystem fileSystem = new();
+            if (IsDifferentDirectory)
+            {
+                fileSystem.AddDirectory("test-configs");
+            }
+
+            if (isAbsolutePath)
+            {
+                userProvidedConfigFilePath = fileSystem.Path.GetFullPath(userProvidedConfigFilePath);
+                finalConfigFilePath = fileSystem.Path.GetFullPath(finalConfigFilePath);
+            }
+
+            if (environmentFile is not null)
+            {
+                fileSystem.AddEmptyFile(environmentFile);
+            }
+
+            FileSystemRuntimeConfigLoader runtimeConfigLoader;
+            if (userProvidedConfigFilePath is not null)
+            {
+                runtimeConfigLoader = new(fileSystem, userProvidedConfigFilePath);
+            }
+            else
+            {
+                runtimeConfigLoader = new(fileSystem);
+            }
+
+            Assert.AreEqual(finalConfigFilePath, runtimeConfigLoader.ConfigFilePath);
```

Comment:
Will `runtimeConfigLoader.ConfigFilePath` always be a full path? this test seems to indicate that it could be either an absolute path, a relative path, or just a file name.